### PR TITLE
Add detail to slow cluster state warning message

### PR DESCRIPTION
**Pull request by DaveCTurner**
Today if writing the cluster state is slow then we report how many index
metadata documents we wrote but we don't mention how many documents were
deleted. Deleting docs might trigger a merge and explain why the update
is slower than expected.

This commit adds info about the number of removed indices and also
distinguishes new indices from updates to existing indices, since they
also have different behaviour in terms of deletes.

### CODE REVIEW

**Review by idegtiarenko — APPROVED** [review]
:+1:

File: server/src/main/java/org/elasticsearch/gateway/PersistedClusterStateService.java [hunk 1]
```diff
@@ -1056,15 +1049,31 @@ public void close() throws IOException {
...
-        static class WriterStats {
-            final boolean globalMetaUpdated;
-            final long numIndicesUpdated;
-            final long numIndicesUnchanged;
-
-            WriterStats(boolean globalMetaUpdated, long numIndicesUpdated, long numIndicesUnchanged) {
-                this.globalMetaUpdated = globalMetaUpdated;
-                this.numIndicesUpdated = numIndicesUpdated;
-                this.numIndicesUnchanged = numIndicesUnchanged;
+        static record WriterStats(
```
Comment (arteam) [inline server/src/main/java/org/elasticsearch/gateway/PersistedClusterStateService.java:1052]: I believe records are static by default, so the `static` modifier seems redundant to me here.

Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/gateway/PersistedClusterStateService.java:1052]: ++ I opened https://github.com/elastic/elasticsearch/issues/83311.

**Review by arteam — APPROVED** [review]
LGTM! Thanks David!

**Pull request merged**